modules/browserHelper.js

```
@@ -0,0 +1,191 @@
+                      let bgUrl = window.getComputedStyle(tags[i]).backgroundImage;
+                      bgUrl = extractUrl(bgUrl);
+                      if (bgUrl !== '') {
+                        allBackgroundURLs.push(bgUrl);
+                      }
+                    }
+                    return allBackgroundURLs;
+                  }
+
+                  function add_images() {
+                    let urls = backgroundUrls();
+                    for (let i = 0; i < urls.length; i++) {
+                      let img = document.createElement('img');
+                      img.src = urls[i];
+                      document.body.appendChild(img);
+                    }
+                    return urls.length;
+                  }
+
+                  return add_images();`;
+  return executeScript(script);
+};
+
+const removeBackgroundImagesAddedAsNewImages = numberOfImages => {
+  const script = ` var images = document.querySelectorAll('img');
+                    for( var i = 1; i <= ${numberOfImages}; i++ ) {
+                      document.body.removeChild(images[images.length - i]);
+                    }
+                 `;
+
+  return executeScript(script);
+};
+
+const imagesLoaded = () => {
+  if (CURRENT_BROWSER !== SUPPORTED_BROWSERS.SAFARI) {
+    return checkAllImageTagsLoaded();
+  }
+  let loaded = false;
+
+  return addBackgroundImagesAsNewImages()
+    .then(numberOfNewImagesAdded => {
+      return checkAllImageTagsLoaded()
+        .then(allImagesLoaded => {
+          loaded = allImagesLoaded;
+          return removeBackgroundImagesAddedAsNewImages(numberOfNewImagesAdded);
+        })
+        .then(() => {
+          return loaded;
+        });
+    });
+};
+
+const checkImagesLoaded = () => {
+  return expect(imagesLoaded()).to.eventually.be.true;
+};
+
+const getBrowserConfig = () => {
+  const browserCapabilities = browser.desiredCapabilities;
+  const browserConfig = {
+    // Maybe fail if these values are not provided
```

Have you made a decision about if you want to throw an error here?

---

In fact... why does this function exist? Why not just use `browser.desiredCapabilities` directly? The only difference I can see is renaming "os_version" to "osVersion" but this is just for metadata so I'm not sure if that matters